linetools/spectra/convolve.py

```diff
@@ -0,0 +1,175 @@
+    else:
+        try:
+            abs(int(edge))
+        except TypeError:
+            raise ValueError('Unknown value for edge keyword: %s' % edge)
+        med1 = np.median(a[:edge])
+        med2 = np.median(a[-edge:])
+        temp1 =  med1 * np.ones(n) , a, med2 * np.ones(n)
+
+    temp2 = np.convolve(np.concatenate(temp1), window, mode=str('same'))
+
+    return temp2[n:-n]
+
+'''
+def convolve_constant_dv(wa, fl, wa_dv=None, npix=4., vfwhm=None, dv_const=False):
+    """ Convolve a wavelength array with a gaussian of constant
+    velocity width.
+
+    If `vfwhm` is specified an intermediate wavelength array with
+    constant velocity pixel width is calculated. Otherwise, both
+    `wa_dv` and `npix` must be given -- this is faster because no
+    intermediate array needs to be calculated.
+
+    Parameters
+    ----------
+    fl, wa : arrays of floats, length N
+      The array to be convolved and its wavelengths.
+    vfwhm : float, optional
+      Full width at half maximum in velocity space (km/s) of the
+      gaussian kernel with which to convolve `fl`.
+    npix : float, default 4
+      Number of pixels corresponding to `vfwhm` in `wa_dv` if given,
+      otherwise `wa` is interpolated to an array with velocity pixel
+      width = vfwhm / npix.
+    wa_dv : array of floats, default `None`
+      Wavelength array with a constant velocity width (this can be
+      generated with make_constant_dv_wa_scale()).
+    dv_const : Bool, default 'False'
+      Specifies whether the input array has constant velocity
+
+    Returns
+    -------
+    fl_out : array of length N
+      fl convolved with the gaussian kernel with the specified FWHM.     
+
+    Examples
+    --------
+    >>> from barak import sed
+    >>> wa = np.arange(5000, 7000, 0.03)
+    >>> fl = np.random.randn(len(wa)) + 1.
+    >>> instrument_profile_fwhm = 5.5 # resolution fwhm km/s
+    >>> flsmooth = convolve_constant_dv(wa, fl, vfwhm=instrument_profile)
+
+    To avoid generating a new constant velocity scale for multiple
+    convolutions of the same spectrum, you can create your own and
+    pass it as the wa_dv argument:
+
+    >>> from barak import sed
+    >>> subpixkms = 1.3  # km/s    
+    >>> wa_dv = sed.make_constant_dv_wa_scale(wa[0], wa[-1], subpixkms)
+    >>> npix = instrument_profile / subpixkms
+    >>> flsmooth = convolve_constant_dv(wa, fl, wa_dv=wa_dv, npix=npix)
+    """
+    # interpolate to the log-linear scale, convolve, then
+    # interpolate back again.
+    if vfwhm is not None:
+        wa_dv = make_constant_dv_wa_scale(wa[0], wa[-1], float(vfwhm)/npix)
+    if dv_const is False:
+        fl_dv = np.interp(wa_dv, wa, fl)
+    else: fl_dv = fl
+    # Convolve
+    fl_dv_smoothed = convolve_psf(fl_dv, npix)
+    # Interpolate as needed
+    if dv_const is False:
+        fl_out = np.interp(wa, wa_dv, fl_dv_smoothed)
+    else: fl_out = fl_dv_smoothed
+    # Return
+    return fl_out
+'''
```

Comment:
It think it's better to remove this (convolve_constant_dv) instead of commenting it out.  It's always easy to add it in later, right?